### DISCUSSION TITLE
Problem with logs

### DISCUSSION THREAD

**Discussion by graemev** [forwardemail/forwardemail.net#271] [General]
I have a lot of difficulty with the "logs" on the website.

It was not clear what the difference between the 2 logs was. I finally decided "logs" was a log about incoming email (MX) and "mail" was a log about outgoing email (SMTP) .

But the entry "logs" is many many hours out of date. Overnight I noted all my mail was tagged as     550error:1C800064:Provider routines::bad decrypt ...I I look at the logs right now (09:15 BST) this is all they show (550error). However I just sent a test mail via proton email and it arrived just fine. There is no entry in the log (yet) as the latest it has is 02:48 BST . It is very difficult to debug this with logs so old.

So my "guess" is , it **was** broken, it's not now, but I can't see when it changed.

**Comment by titanism** [comment]
Hi there @graemev 👋   There was an issue over the past 24 hours with our encryption/decryption helper, where it was using the wrong method.  We rolled back to use AES-256-CBC (as we did before) instead of a new approach which used ChaChaPoly-1305.  There were a few edge cases where you might have seen a decrypt error as shown above in your post and in your log entries.  Rest assured this issue is resolved and emails will flow through without issues.  It was just a temporary problem.

**Reply by graemev** [reply]
You did not address why the logs (the main point of the post) are so very out of date? Right now its just after noon and the latest log is from 06:00

**Comment by titanism** [comment]
Logs are stored and then parsed on an interval every 5 minutes.  Only one unique log is stored per message every hour.

<https://github.com/forwardemail/forwardemail.net/blob/e36a2c1cf75768d34bad57c709241e964bfdbc27/jobs/index.js#L151-L155>

<https://github.com/forwardemail/forwardemail.net/blob/master/jobs/parse-logs.js>

**Reply by titanism** [reply]
Also @graemev, these logs are for errors only.

**Comment by graemev** [comment]
It's 12:40 BST ...here is the top of my "logs" on a screenshot

(these are all admin mails set during the night, mailtest, updates from my Humax PVR etc

(AFYI, I see no "errors" here)

![Forwardemai-logs-Screenshot_20240803_123819](https://github.com/user-attachments/assets/2b90ec82-cddd-4df8-ad53-00ee6aab9371)

The log called "mail" is similarly out of date:
![forwardemail-email-log-Screenshot_20240803_124328](https://github.com/user-attachments/assets/e0526ce1-c559-4ebb-8d74-b8ca7da5fa9e)

**Reply by graemev** [reply]
Maybe delete my "images" once you seen/grabbed them ...rather than leave around for bots to harvest

**Reply by titanism** [reply]
@graemev With the exception of logs created for when outbound SMTP email is queued/created/sent, the only other logs shown are for MX errors.
